### PR TITLE
fix: use cowswap cf-pages preview in widget builds

### PR DESCRIPTION
**Pull request by alfetopito**
# Summary

Use the new cf-pages preview from CoW Swap in widget preview builds.
Long and weird branch names are also supported.

# To Test

1. Open the widget preview
* Should load CoW Swap preview from cf-pages and point to this PR's build https://fix-cf-widget-url.swap-dev-5u6.pages.dev/

Additionally, created PRs from branches with edge case names to assert it worked. See https://github.com/cowprotocol/cowswap/pull/7677 and https://github.com/cowprotocol/cowswap/pull/7676

## Summary by CodeRabbit

* **Tests**
  * Added Jest coverage for branch-name normalization and Cloudflare Pages subdomain conversion, including edge cases and fallback behavior.
* **Refactor**
  * Updated the preview environment URL generation to use Cloudflare Pages preview subdomains instead of constructing a Vercel hostname from a prior suffix.
  * Improved branch key derivation by normalizing branch names and translating Vercel preview slugs.
* **Chores**
  * Injected the branch reference environment value into the client build for more reliable preview labeling.

### CODE REVIEW

**Comment by alfetopito** [comment]
@coderabbitai review

File: apps/widget-configurator/src/app/configurator/hooks/useWidgetParamsAndSettings.ts [hunk 1]
```diff
@@ -5,7 +5,26 @@ import { CowSwapWidgetParams, TradeType, WidgetHookEvents } from '@cowprotocol/w
...
+export const branchNameToCfPagesSubdomain = (branchName: string): string => {
+  return branchName
+    .toLowerCase()
+    .replace(/[^a-z0-9]/g, '-')
+    .slice(0, cfPagesPreviewSubdomainMaxLength)
+}
+
+export const vercelPreviewSlugToCfPagesSubdomain = (
+  branchSlug: string,
+  branchName = process.env.VERCEL_GIT_COMMIT_REF,
```
Comment (alfetopito) [inline apps/widget-configurator/src/app/configurator/hooks/useWidgetParamsAndSettings.ts:26]: New trick!
Uses - if available, but I expect it always will be - the branch name from the build, which avoids cases where the branch name is unrecoverable from the url, such as:
- Widget https://widget-configurator-git-fix-deepsec-high-fro-87dcc4-cowswap-dev.vercel.app/
- Vs CoW Swap https://fix-deepsec-high-frontend-ha.swap-dev-5u6.pages.dev/
- Vs branch name `fix/deepsec-high-frontend-hardening`

File: apps/widget-configurator/src/app/configurator/hooks/useWidgetParamsAndSettings.ts [hunk 2]
```diff
@@ -5,7 +5,26 @@ import { CowSwapWidgetParams, TradeType, WidgetHookEvents } from '@cowprotocol/w
...
+const cfPagesPreviewSuffix = `.swap-dev-5u6.pages.dev`
+const cfPagesPreviewSubdomainMaxLength = 28
+const vercelPreviewHashSuffix = /-[a-f0-9]{6}$/
+
+export const branchNameToCfPagesSubdomain = (branchName: string): string => {
+  return branchName
+    .toLowerCase()
+    .replace(/[^a-z0-9]/g, '-')
+    .slice(0, cfPagesPreviewSubdomainMaxLength)
+}
```
[the remark this answers is not in the document]
Comment (alfetopito) [reply apps/widget-configurator/src/app/configurator/hooks/useWidgetParamsAndSettings.ts:22]: Sure it's needed? Tested with a pretty weird branch name and it worked fine https://github.com/cowprotocol/cowswap/pull/7676

Any suggestions for a branch name that would break this?

[the remark this answers is not in the document]
Comment (alfetopito) [reply apps/widget-configurator/src/app/configurator/hooks/useWidgetParamsAndSettings.ts:22]: Fixed on https://github.com/cowprotocol/cowswap/pull/7674/commits/04d44cfca68217fab9b92a5f7170942a39a9632d

**Review by elena-zh — APPROVED** [review]
Thank you

**Review by Danziger — APPROVED** [review]
